internal/tss/utils/transaction_service.go

```diff
@@ -0,0 +1,233 @@
+	}
+	var errorResult xdr.TransactionResult
+	_, err = xdr3.Unmarshal(bytes.NewReader(decodedBytes), &errorResult)
+	if err != nil {
+		return tss.RPCTXCode{OtherCodes: tss.UnMarshalBinaryCode}, fmt.Errorf(unMarshalErr, errorResultXdr)
+	}
+	return tss.RPCTXCode{
+		TxResultCode: errorResult.Result.Code,
+	}, nil
+}
+
+func (t *transactionService) sendRPCRequest(method string, params map[string]string) (tss.RPCResponse, error) {
+	payload := map[string]interface{}{
+		"jsonrpc": "2.0",
+		"id":      1,
+		"method":  method,
+		"params":  params,
+	}
+	jsonData, err := json.Marshal(payload)
+
+	if err != nil {
+		return tss.RPCResponse{}, fmt.Errorf("marshaling payload")
+	}
+
+	resp, err := t.HTTPClient.Post(t.RPCURL, "application/json", bytes.NewBuffer(jsonData))
+	if err != nil {
+		return tss.RPCResponse{}, fmt.Errorf("%s: sending POST request to rpc: %v", method, err)
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return tss.RPCResponse{}, fmt.Errorf("%s: unmarshaling RPC response", method)
+	}
+	var res tss.RPCResponse
+	err = json.Unmarshal(body, &res)
+	if err != nil {
+		return tss.RPCResponse{}, fmt.Errorf("%s: parsing RPC response JSON", method)
+	}
+	if res.RPCResult == (tss.RPCResult{}) {
+		return tss.RPCResponse{}, fmt.Errorf("%s: response missing result field", method)
+	}
+	return res, nil
+}
+
+func (t *transactionService) SendTransaction(transactionXdr string) (tss.RPCSendTxResponse, error) {
+	rpcResponse, err := t.sendRPCRequest("sendTransaction", map[string]string{"transaction": transactionXdr})
+	sendTxResponse := tss.RPCSendTxResponse{}
+	sendTxResponse.TransactionXDR = transactionXdr
+	if err != nil {
+		sendTxResponse.Code.OtherCodes = tss.RPCFailCode
+		return sendTxResponse, fmt.Errorf("RPC fail: %s", err.Error())
+	}
+	sendTxResponse.Status = tss.RPCTXStatus(rpcResponse.RPCResult.Status)
+	sendTxResponse.Code, err = t.parseErrorResultXDR(rpcResponse.RPCResult.ErrorResultXDR)
+	sendTxResponse.TransactionHash = rpcResponse.RPCResult.Hash
+	return sendTxResponse, err
+}
+
+func (t *transactionService) GetTransaction(transactionHash string) (tss.RPCGetIngestTxResponse, error) {
+	rpcResponse, err := t.sendRPCRequest("getTransaction", map[string]string{"hash": transactionHash})
+	if err != nil {
+		return tss.RPCGetIngestTxResponse{Status: tss.ErrorStatus}, fmt.Errorf("RPC Fail: %s", err.Error())
+	}
+	getIngestTxResponse := tss.RPCGetIngestTxResponse{}
+	getIngestTxResponse.Status = tss.RPCTXStatus(rpcResponse.RPCResult.Status)
+	getIngestTxResponse.EnvelopeXDR = rpcResponse.RPCResult.EnvelopeXDR
+	getIngestTxResponse.ResultXDR = rpcResponse.RPCResult.ResultXDR
+	if getIngestTxResponse.Status != tss.NotFoundStatus {
+		getIngestTxResponse.CreatedAt, err = strconv.ParseInt(rpcResponse.RPCResult.CreatedAt, 10, 64)
+		if err != nil {
+			return tss.RPCGetIngestTxResponse{Status: tss.ErrorStatus}, fmt.Errorf("unable to parse createAt: %s", err.Error())
```

Comment:
For error string formatting you can use the `%w` so you can pass the error itself without calling `Error()`
```suggestion
			return tss.RPCGetIngestTxResponse{Status: tss.ErrorStatus}, fmt.Errorf("unable to parse createdAt: %w", err)
```

Reply:
done